src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py

```diff
@@ -314,6 +352,13 @@ def Update(self, cmd, resource_group_name, cluster_name, auto_upgrade_minor_vers
                 if fe_ssl_cert_file and fe_ssl_key_file:
                     self.__set_inference_ssl_from_file(configuration_protected_settings, fe_ssl_cert_file, fe_ssl_key_file)
 
+        # if no entries are existed in configuration_protected_settings, configuration_settings, return whatever passed
+        #  in the Update function(empty dict or None).
+        if len(configuration_settings) == 0:
+            configuration_settings = input_configuration_settings
+        if len(configuration_protected_settings) == 0:
+            configuration_protected_settings = input_configuration_protected_settings
+
         return PatchExtension(auto_upgrade_minor_version=auto_upgrade_minor_version,
```

Comment:
You are not updating the configuration settings or configuration_protected_settings in the patch extension that you are returning. Is that intentional?

Reply:
We will not update it only if configuration settings contains nothing and respect whatever passed into the update method, empty dict or none.